src/models/callercalleemodel.cpp

```diff
@@ -0,0 +1,138 @@
+
+  Copyright (C) 2017 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Nate Rogers <nate.rogers@kdab.com>
+
+  Licensees holding valid commercial KDAB Hotspot licenses may use this file in
+  accordance with Hotspot Commercial License Agreement provided with the Software.
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "callercalleemodel.h"
+
+CallerCalleeModel::CallerCalleeModel(QObject* parent)
+    : QAbstractTableModel(parent)
+{
+}
+
+CallerCalleeModel::~CallerCalleeModel() = default;
+
+QVariant CallerCalleeModel::headerData(int section, Qt::Orientation orientation, int role) const
+{
+    if (role != Qt::DisplayRole || orientation != Qt::Horizontal) {
+        return {};
+    }
+    if (role == Qt::InitialSortOrderRole) {
+        if (section == SelfCost || section == InclusiveCost)
+        {
+            return Qt::DescendingOrder;
+        }
+    }
+    switch (static_cast<Columns>(section)) {
+        case Symbol:
+            return tr("Symbol");
+        case Binary:
+            return tr("Binary");
+        case Location:
+            return tr("Location");
+        case Address:
+            return tr("Address");
+        case SelfCost:
+            return tr("Self Cost");
+        case InclusiveCost:
+            return tr("Inclusive Cost");
+        case NUM_COLUMNS:
+            // fall-through
+            break;
+    }
+
+    return {};
+}
+
+QVariant CallerCalleeModel::data(const QModelIndex& index, int role) const
+{
+    const auto* item = &m_rows.at(index.row());
+
+    if (!item) {
+        return {};
+    }
+
+    if (role == SortRole) {
+        switch (static_cast<Columns>(index.column())) {
+            case Symbol:
+                return item->symbol;
+            case Binary:
+                return item->binary;
+            case Location:
+                return item->location;
+            case Address:
+                return item->address;
+            case SelfCost:
+                return item->selfCost;
+            case InclusiveCost:
+                return item->inclusiveCost;
+            case NUM_COLUMNS:
+                // do nothing
+                break;
+        }
+    } else if (role == FilterRole) {
+        return item->symbol + item->binary + item->location;
```

Comment:
also, wrap it in a `QString(...)` to make the conversion from QStringBuilder explicit. I just pushed some changes which will make this necessary